src/NHSD.BuyingCatalogue.Identity.UserDatabase/Deployment/CreateTestUsers.sql

```diff
@@ -1,11 +1,18 @@
 ﻿DECLARE @aliceEmail AS nvarchar(50) = N'AliceSmith@email.com';
 DECLARE @bobEmail AS nvarchar(50) = N'BobSmith@email.com';
 
-IF NOT EXISTS(
+IF '$(ASPNETCORE_ENVIRONMENT)' <> 'Production' 
```

Comment:
No problems with this change. Just had a thought whether or not it would be useful to do something like the appsettings.development.json file. So we only load the SQL file depending on the `ASPNETCORE_ENVIRONMENT` environment variable. This would remove the if condition & keep the concerns of development and production separate. What do you think? Definitely not saying do this now.

Reply:
That makes sense although there's no easy way to do that that I'm aware of. I think we'd need to play around with the build config and make sure we pass the environment variable to MSBuild.

I guess we can leave like this for now and create a new task for that. Although it's not elegant at least this will prevent the test users from being created in production for now.

Reply:
Is it possible instead to have this as something like INSERT_TEST_DATA, so it's clear what the intention is of the sqlcmd variable?
Also, might be worth amending the user guard, so if any are set it doesn't insert them.

Reply:
I was thinking something could be done within the `configure-db.sh` to look for a file with the appropriate environment variable value. But sounds like Stuart would like to change the context of the variable anyways.

Reply:
I just stuck with `ASPNETCORE_ENVIRONMENT` as we already had that for the .NET environments but I did originally think an additional environment variable might have been preferable, particularly as the DB isn't an ASP.NET Core environment. Therefore, I'm happy to go with Stuart's suggestion of INSERT_TEST_DATA.

@stuart-bass-bjss can you clarify what you mean by amending the guard? Currently, the test users won't be inserted if they already exist.

@sam-roberts-bjss when we deploy the dacpac using DacFx we shouldn't have any need for `configure-db.sh`. Ideally I'd like to look into whether we can tweak the docker build to make use of DacFx at some point, if time permits (tech debt).

Reply:
I meant to amend the guard, so the test users aren't inserted if any users exist.

Reply:
Ah, that's fine. That guard is already there.

Reply:
Or perhaps not. To clarify: do mean any *test* users or *any* users?